### PR TITLE
[STAT] Update Static Analysis docs

### PR DESCRIPTION
**Pull request by dastrong**
### What does this PR do? What is the motivation?

This PR adds documentation for all our new JS/TS rulesets and their rules.

- new rulesets were added, so I've added them to the pull-config files
- updated frontmatter and the body sections of our rules index page
- added language headings to group rulesets better
- updated the beta banner to include all current supported languages

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 1]
```diff
@@ -478,16 +1244,102 @@ further_reading:
 
 Datadog Static Analysis provides out-of-the-box rules to help detect violations in your CI/CD pipelines in code reviews and identify bugs, security, and maintainability issues. For more information, see the [Static Analysis documentation][1].
 
-## Rules
+## Docker Rules
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:1247]:
```suggestion
## Docker rules
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 2]
```diff
@@ -478,16 +1244,102 @@ further_reading:
...
 
 **Ruleset ID:** `docker-best-practices`
 
+Best practices for using Docker.
+
 {{< sa-rule-list "docker_best_practices_data" >}}
 
 <br>
 
+## JavaScript Rules
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:1259]:
```suggestion
## JavaScript rules
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 3]
```diff
@@ -478,16 +1244,102 @@ further_reading:
...
+
+**Ruleset ID:** `jsx-react`
+
+This plugin exports a `recommended` configuration that enforces React good practices.
+
+{{< sa-rule-list "jsx_react_data" >}}
+
+<br>
+
+## Python Rules
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:1341]:
```suggestion
## Python rules
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 4]
```diff
@@ -580,6 +1432,89 @@ Rules focused on finding security and vulnerability issues in your Python code,
 
 <br>
 
+## TypeScript Rules
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:1435]:
```suggestion
## TypeScript rules
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 5]
```diff
@@ -456,6 +827,401 @@ python_security_data:
...
+    text: "Avoid reassigning exceptions in catch clauses"
+  - link: "/continuous_integration/static_analysis/rules/typescript-best-practices/no-explicit-any"
+    tag: "no-explicit-any"
+    text: "Avoid the any type"
+  - link: "/continuous_integration/static_analysis/rules/typescript-best-practices/no-extra-non-null-assertion"
+    tag: "no-extra-non-null-assertion"
+    text: "Avoid extra non-null assertions"
+  - link: "/continuous_integration/static_analysis/rules/typescript-best-practices/no-implied-eval"
+    tag: "no-implied-eval"
+    text: "Prevent the use of eval()-like methods"
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:927]:
```suggestion
    text: "Prevent the use methods similar to eval()"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 6]
```diff
@@ -67,6 +67,377 @@ docker_best_practices_data:
...
+    text: "Avoid new statements with the Symbol object"
+  - link: "/continuous_integration/static_analysis/rules/javascript-best-practices/no-octal"
+    tag: "no-octal"
+    text: "Avoid using octal literals to prevent unexpected behavior"
+  - link: "/continuous_integration/static_analysis/rules/javascript-best-practices/no-proto"
+    tag: "no-proto"
+    text: "Avoid the use of the __proto__ property"
+  - link: "/continuous_integration/static_analysis/rules/javascript-best-practices/no-script-url"
+    tag: "no-script-url"
+    text: "Avoid using javascript in URLs"
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:154]:
```suggestion
    text: "Avoid using JavaScript in URLs"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 7]
```diff
@@ -67,6 +67,377 @@ docker_best_practices_data:
...
+    text: "Avoid empty character classes in regular expressions"
+  - link: "/continuous_integration/static_analysis/rules/javascript-best-practices/no-empty-pattern"
+    tag: "no-empty-pattern"
+    text: "Avoid empty destructuring patterns"
+  - link: "/continuous_integration/static_analysis/rules/javascript-best-practices/no-ex-assign"
+    tag: "no-ex-assign"
+    text: "Avoid reassigning exceptions in catch clauses"
+  - link: "/continuous_integration/static_analysis/rules/javascript-best-practices/no-implied-eval"
+    tag: "no-implied-eval"
+    text: "Prevent the use of eval()-like methods"
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:130]:
```suggestion
    text: "Prevent the use of methods similar to eval()"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 8]
```diff
@@ -456,6 +827,401 @@ python_security_data:
...
+    text: "RegExp with non-literal values it might allow a DOS attack"
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/detect-non-literal-require"
+    tag: "detect-non-literal-require"
+    text: "Avoid require with non-literal values"
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/detected-jwt-token"
+    tag: "detected-jwt-token"
+    text: "Detects hardcoded JWT tokens within the codebase."
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/hardcoded-hmac-key"
+    tag: "hardcoded-hmac-key"
+    text: "Detects hardcoded hmac keys"
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:1200]:
```suggestion
    text: "Detects hardcoded HMAC keys"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 9]
```diff
@@ -67,6 +67,377 @@ docker_best_practices_data:
...
+    text: "RegExp with non-literal values it might allow a DOS attack"
+  - link: "/continuous_integration/static_analysis/rules/javascript-node-security/detect-non-literal-require"
+    tag: "detect-non-literal-require"
+    text: "Avoid require with non-literal values"
+  - link: "/continuous_integration/static_analysis/rules/javascript-node-security/detected-jwt-token"
+    tag: "detected-jwt-token"
+    text: "Detects hardcoded JWT tokens within the codebase."
+  - link: "/continuous_integration/static_analysis/rules/javascript-node-security/hardcoded-hmac-key"
+    tag: "hardcoded-hmac-key"
+    text: "Avoid hardcoded hmac keys"
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:379]:
```suggestion
    text: "Avoid hardcoded HMAC keys"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 10]
```diff
@@ -67,6 +67,377 @@ docker_best_practices_data:
...
+    text: "Avoid `eval` with expressions"
+  - link: "/continuous_integration/static_analysis/rules/javascript-node-security/detect-new-buffer"
+    tag: "detect-new-buffer"
+    text: "Avoid Buffer(argument) with non-literal values"
+  - link: "/continuous_integration/static_analysis/rules/javascript-node-security/detect-non-literal-fs-filename"
+    tag: "detect-non-literal-fs-filename"
+    text: "Avoid variables in 'fs' calls filename argument"
+  - link: "/continuous_integration/static_analysis/rules/javascript-node-security/detect-non-literal-regexp"
+    tag: "detect-non-literal-regexp"
+    text: "RegExp with non-literal values it might allow a DOS attack"
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:370]:
```suggestion
    text: "Regular expressions with non-literal values might allow a DOS attack"
```

Comment (buraizu) [reply content/en/continuous_integration/static_analysis/rules/_index.md:370]: Could you confirm if this is meant to say `DDOS attack`?

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 11]
```diff
@@ -67,6 +67,377 @@ docker_best_practices_data:
...
+    text: "Avoid command injection"
+  - link: "/continuous_integration/static_analysis/rules/javascript-node-security/crypto-avoid-weak-hash"
+    tag: "crypto-avoid-weak-hash"
+    text: "Avoid weak hash algorithm from CryptoJS"
+  - link: "/continuous_integration/static_analysis/rules/javascript-node-security/detect-buffer-noassert"
+    tag: "detect-buffer-noassert"
+    text: "Avoid calls to 'buffer' with 'noAssert' flag set"
+  - link: "/continuous_integration/static_analysis/rules/javascript-node-security/detect-child-process"
+    tag: "detect-child-process"
+    text: "Avoid instances of 'child_process' & non-literal 'exec()'"
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:358]:
```suggestion
    text: "Avoid instances of 'child_process' and non-literal 'exec()'"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 12]
```diff
@@ -456,6 +827,401 @@ python_security_data:
...
+    text: "Avoid command injection"
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/crypto-avoid-weak-hash"
+    tag: "crypto-avoid-weak-hash"
+    text: "Avoid weak hash algorithm from CryptoJS"
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/detect-buffer-noassert"
+    tag: "detect-buffer-noassert"
+    text: "Avoid calls to 'buffer' with 'noAssert' flag set"
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/detect-child-process"
+    tag: "detect-child-process"
+    text: "Avoid instances of 'child_process' & non-literal 'exec()'"
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:1179]:
```suggestion
    text: "Avoid instances of 'child_process' and non-literal 'exec()'"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 13]
```diff
@@ -456,6 +827,401 @@ python_security_data:
...
+    text: "Avoid using TSLint comments"
+  - link: "/continuous_integration/static_analysis/rules/typescript-code-style/class-name"
+    tag: "class-name"
+    text: "Class name should be PascalCase"
+  - link: "/continuous_integration/static_analysis/rules/typescript-code-style/eqeqeq"
+    tag: "eqeqeq"
+    text: "Enforce the use of === and !=="
+  - link: "/continuous_integration/static_analysis/rules/typescript-code-style/func-name-matching"
+    tag: "func-name-matching"
+    text: "Function names must match the name of the assignation."
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:1016]:
```suggestion
    text: "Function names must match the name of the assignation"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 14]
```diff
@@ -456,6 +827,401 @@ python_security_data:
...
+    text: "Avoid non-null assertion in confusing locations"
+  - link: "/continuous_integration/static_analysis/rules/typescript-code-style/no-div-regex"
+    tag: "no-div-regex"
+    text: "Avoid equal signs explicitly at the beginning of regex"
+  - link: "/continuous_integration/static_analysis/rules/typescript-code-style/no-duplicate-imports"
+    tag: "no-duplicate-imports"
+    text: "Avoid duplicate module imports"
+  - link: "/continuous_integration/static_analysis/rules/typescript-code-style/no-empty-interface"
+    tag: "no-empty-interface"
+    text: "Avoid the declaration of empty interfaces."
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:1049]:
```suggestion
    text: "Avoid the declaration of empty interfaces"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 15]
```diff
@@ -456,6 +827,401 @@ python_security_data:
...
+    text: "Avoid Object constructors"
+  - link: "/continuous_integration/static_analysis/rules/typescript-code-style/no-return-assign"
+    tag: "no-return-assign"
+    text: "Avoid assignment operators in return statements"
+  - link: "/continuous_integration/static_analysis/rules/typescript-code-style/no-self-compare"
+    tag: "no-self-compare"
+    text: "Avoid comparisons where both sides are exactly the same"
+  - link: "/continuous_integration/static_analysis/rules/typescript-code-style/no-useless-empty-export"
+    tag: "no-useless-empty-export"
+    text: "Avoid empty exports that don't change anything."
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:1079]:
```suggestion
    text: "Avoid empty exports that don't change anything"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 16]
```diff
@@ -456,6 +827,401 @@ python_security_data:
...
+    text: "Avoid variables in 'fs' calls filename argument"
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/detect-non-literal-regexp"
+    tag: "detect-non-literal-regexp"
+    text: "RegExp with non-literal values it might allow a DOS attack"
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/detect-non-literal-require"
+    tag: "detect-non-literal-require"
+    text: "Avoid require with non-literal values"
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/detected-jwt-token"
+    tag: "detected-jwt-token"
+    text: "Detects hardcoded JWT tokens within the codebase."
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:1197]:
```suggestion
    text: "Detects hardcoded JWT tokens within the codebase"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 17]
```diff
@@ -456,6 +827,401 @@ python_security_data:
...
+    text: "Avoid `eval` with expressions"
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/detect-new-buffer"
+    tag: "detect-new-buffer"
+    text: "Avoid Buffer(argument) with non-literal values"
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/detect-non-literal-fs-filename"
+    tag: "detect-non-literal-fs-filename"
+    text: "Avoid variables in 'fs' calls filename argument"
+  - link: "/continuous_integration/static_analysis/rules/typescript-node-security/detect-non-literal-regexp"
+    tag: "detect-non-literal-regexp"
+    text: "RegExp with non-literal values it might allow a DOS attack"
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:1191]:
```suggestion
    text: "Regular expressions with non-literal values might allow a DOS attack"
```

File: content/en/continuous_integration/static_analysis/rules/_index.md [hunk 18]
```diff
@@ -67,6 +67,377 @@ docker_best_practices_data:
...
+    text: "Enforce a maximum number of parameters in a function"
+  - link: "/continuous_integration/static_analysis/rules/javascript-code-style/method-name"
+    tag: "method-name"
+    text: "Method name should use camelCase"
+  - link: "/continuous_integration/static_analysis/rules/javascript-code-style/no-array-constructor"
+    tag: "no-array-constructor"
+    text: "Avoid Array constructors"
+  - link: "/continuous_integration/static_analysis/rules/javascript-code-style/no-div-regex"
+    tag: "no-div-regex"
+    text: "Avoid equal signs explicitly at the beginning of regex"
```
Comment (buraizu) [inline content/en/continuous_integration/static_analysis/rules/_index.md:231]:
```suggestion
    text: "Avoid equal signs explicitly at the beginning of regular expressions"
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks, these updates are mostly minor, but just flagging one section with a question

File: content/en/continuous_integration/static_analysis/rules/_index.md (hunk 10 above)
Comment (dastrong) [reply content/en/continuous_integration/static_analysis/rules/_index.md:370]: Over the limit of 60 characters. I've updated to `Detects non-literal values in regular expressions`

File: content/en/continuous_integration/static_analysis/rules/_index.md (hunk 17 above)
Comment (dastrong) [reply content/en/continuous_integration/static_analysis/rules/_index.md:1191]: Same.

Over the limit of 60 characters. I've updated to `Detects non-literal values in regular expressions`

**Review by buraizu — APPROVED** [review]

**Pull request merged**